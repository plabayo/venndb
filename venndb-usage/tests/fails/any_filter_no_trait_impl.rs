use venndb::VennDB;

#[derive(Debug, VennDB)]
struct Employee {
    #[venndb(filter, any)]
    department: Department,
}

#[derive(Debug, PartialEq, Eq, Hash, Clone)]
pub enum Department {
    Any,
    Engineering,
    Sales,
    Marketing,
    HR,
}

fn main() {
    let _ = EmployeeDB::new();
}
