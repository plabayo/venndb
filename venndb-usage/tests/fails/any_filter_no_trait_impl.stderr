error[E0277]: the trait bound `Department: venndb::Any` is not satisfied
 --> tests/fails/any_filter_no_trait_impl.rs:3:17
  |
3 | #[derive(Debug, VennDB)]
  |                 ^^^^^^ the trait `venndb::Any` is not implemented for `Department`
  |
  = help: the following other types implement trait `venndb::Any`:
            Box<T>
            Rc<T>
            Arc<T>
            Option<T>
            &T
  = note: this error originates in the derive macro `VennDB` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Department: venndb::Any` is not satisfied
 --> tests/fails/any_filter_no_trait_impl.rs:3:17
  |
3 | #[derive(Debug, VennDB)]
  |                 ^^^^^^ the trait `venndb::Any` is not implemented for `Department`, which is required by `&Department: venndb::Any`
  |
  = help: the following other types implement trait `venndb::Any`:
            Box<T>
            Rc<T>
            Arc<T>
            Option<T>
            &T
  = note: required for `&Department` to implement `venndb::Any`
  = note: this error originates in the derive macro `VennDB` (in Nightly builds, run with -Z macro-backtrace for more info)
